File: 1.0/src/cz/mg/vulkan/PFNvkGetDeviceGroupSurfacePresentModesKHX.java
package cz.mg.vulkan;

public class PFNvkGetDeviceGroupSurfacePresentModesKHX extends VkFunctionPointer {
    public PFNvkGetDeviceGroupSurfacePresentModesKHX() {
    }

    public PFNvkGetDeviceGroupSurfacePresentModesKHX(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkGetDeviceGroupSurfacePresentModesKHX(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetDeviceGroupSurfacePresentModesKHX(long value) {
        setValue(value);
    }

    public PFNvkGetDeviceGroupSurfacePresentModesKHX(VkInstance instance) {
        super(instance, new VkString("vkGetDeviceGroupSurfacePresentModesKHX"));
    }

    public void call(VkDevice device, VkSurfaceKHR surface, VkDeviceGroupPresentModeFlagsKHX pModes, VkResult rval){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), surface != null ? surface.getVkAddress() : VkPointer.getNullAddressNative(), pModes != null ? pModes.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getSinkAddressNative());
    }

    protected static native void call(long vkaddress, long device, long surface, long pModes, long rval);
}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkGetDeviceGroupSurfacePresentModesKHX.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkGetDeviceGroupSurfacePresentModesKHX_call(JNIEnv* env, jclass jc, jlong address, jlong device, jlong surface, jlong pModes, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkGetDeviceGroupSurfacePresentModesKHX f = (PFN_vkGetDeviceGroupSurfacePresentModesKHX)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkSurfaceKHR*)jniLongToPointer(surface)),
        ((VkDeviceGroupPresentModeFlagsKHX*)jniLongToPointer(pModes))
    );
}
